www/utils/swrCache.ts

```diff
@@ -0,0 +1,19 @@
+import { Cache, State } from 'swr';
+
+export function localStorageProvider(): Cache {
+  // When initializing, we restore the data from `localStorage` into a map.
+  if (typeof window !== 'undefined') {
+    const map: Map<string, State> = new Map(
+      JSON.parse(localStorage.getItem('app-cache') || '[]')
+    );
+
+    // Before unloading the app, we write back all the data into `localStorage`.
+    window.addEventListener('beforeunload', () => {
+      const appCache = JSON.stringify(Array.from(map.entries()));
+      localStorage.setItem('app-cache', appCache);
+    });
+    return map;
+  }
+
+  return new Map();
+}
```

Comment:
Is there a way to invalidate the data on logout or trigger it manually, don't want to cache stuff in the case of switching accounts

Reply:
Doing a quick test when I log out and log in with a different account it doesn't load the data from the previous account but it does keep it in the storage

<img width="1496" alt="image" src="https://github.com/user-attachments/assets/8a787964-f80a-41da-8e2f-d52d87c6badd" />